compiler/src/iree/compiler/Codegen/Common/GPU/GPUHeuristics.h

```diff
@@ -0,0 +1,52 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "mlir/IR/Types.h"
+
+namespace mlir::iree_compiler {
+
+/// Struct containing information about a matmul's shape and type.
+struct GPUMatmulShapeType {
+  int64_t mSize;
+  int64_t nSize;
+  int64_t kSize;
+  Type aType;
+  Type bType;
+  Type cType;
```

Comment:
Thinking out loud, we might need to rethink this for integer types at some point, e.g. `V_WMMA_I32_16X16X16_IU8` for rdna3: https://www.amd.com/content/dam/amd/en/documents/radeon-tech-docs/instruction-set-architectures/rdna3-shader-instruction-set-architecture-feb-2023_0.pdf

Reply:
I feel the trend is actually more moving towards fp8 or other smaller floating point numbers.. Though integer quantization will remain for sure. People are just circling through these mechanisms anyway; just a matter of time. Will see.